src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  };
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(brute) {
+    if (brute instanceof Herbivore && !brute.hidden) {
+      brute.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(beast => beast.health);
```

Comment:
it works, however, every time `bite` function is called, this filtering gets performed anyway, do you think it's okay?